### ISSUE TITLE
 [ Mod : Click on empty taskbar space ] Sort folders thumbnails by name.

### ISSUE THREAD

**Issue by Inkdecker** [m1lhaus/windhawk-mods#20]
@m1lhaus Hi! I wanted to know whether or not it could  be possible to sort the opened folders thumbnails alphabetically by name uppon double clicking on the taskbar or performing another action? I used to be able to previously do it with 7+ Taskbar Tweaker advanced options by doing a double left click on the folder icon, and it was really usefull to reorganise and keep track of multiples folders oppened in the explorer at once,  but I haven't been able to find a way to replicate this with Windhawk and ExplorerPatcher's Windows 10 taskbar so far.

![folder_sorting_action](https://github.com/user-attachments/assets/5c8e3e5e-06b6-4d95-9f8a-cd912821b73f)

**Comment by m1lhaus** [comment]
Hi @Inkdecker,
do you mean in a similar way the 'Taskbar Thumbnail Reorder' mod is doing it? But automatically via alphabet? I mean it should be possible since the hardest part, detecting the clicked object and reordering thumbnails, is already implemented. However, I think a separate mod would be better for this. Or maybe we could extend "Taskbar Thumbnail Reorder" directly with this feature. What do you think @m417z?

**Comment by Inkdecker** [comment]
Yes this is exactly what i'm talking about! It would be something like combining the clicking actions / actions part of your mod with the ability to rearrangethe thumbnails from "Taskbar Thumbnail Reorder" to make a new mod maybe, I tried to combine both scripts earlier but its way beyond the scope of my coding skill.

**Comment by m1lhaus** [comment]
I might take a look into that. I am interested to see how "Taskbar Thumbnail Reorder" mod does the reordering. Then I will decide whether I go ahead with some implementation. I won't include it in this mod though. I will leave it open for a while as a reminder for the time being.

**Comment by m417z** [comment]
> However, I think a separate mod would be better for this. Or maybe we could extend "Taskbar Thumbnail Reorder" directly with this feature. What do you think @m417z?

I also think so. It could be more appealing to extend "Taskbar Thumbnail Reorder" if it'd be a small addition to the mod, but it's not the case. I'd start by looking at [`MoveItemsFromThumbnail`](https://github.com/ramensoftware/windhawk-mods/blob/2efe8bfdd78e6d3aa282afe913543b2e6e0ac84b/mods/taskbar-thumbnail-reorder.wh.cpp#L452-L478).
